.github/actions/check-api-version-file/action.yml

```diff
@@ -0,0 +1,37 @@
+---
+name: "Check API version file changed"
+description: "Checks that the API version.json file was changed too if the API module was modified"
+inputs:
+  apiDirectory:
+    required: true
+    description: "The directory where the API module is located. Relative to the repo root."
+  apiVersionFile:
+    required: true
+    description: "The path to the JSON file that contains API versioning information, relative to the repo root."
+
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v4
+    - name: Inspect changeset if version file changed
+      shell: bash
+      run: |
+        # main branch is not checked out by default
+        git fetch --no-tags --prune --no-recurse-submodules --depth=1 origin main:main
+
+        # check if there was a change any of the API modules. the last "grep" parses the result of "wc" into a number
+        apiChanges=$(git diff --name-only main || true | grep "${{ inputs.apiDirectory }}" | wc -l | grep -o "[0-9]\+")
+        if [[ $apiChanges -eq 0 ]] 
+          then
+            echo "There were no changes in the API"
+          else 
+            echo "There are $apiChanges changed files in the API, will check if version.json was modified."
+            # if so, check if the changeset also contains the `version.json` file
+            changes=$(git diff --name-only main || true | grep "${{ inputs.apiVersionFile }}" |  wc -l | grep -o "[0-9]\+")
+            if [[ $changes -lt 1 ]]
+              then
+                echo "::error file=./${{ inputs.apiVersionFile }},line=1::Bumping the version is required after a change to the API"
+                exit 1
+            fi
+         fi
```

Comment:
I'm still a bit hesitant about checking the folder diff for understanding if there was an api change, we could rely on the openapi spec files: generating it for the branch and checking the difference with the uploaded on gh-pages ones, this will avoid to get errors for e.g. generic refactor made in the api modules and permit a better separated context verification

Reply:
ok, so we generate the openapi file for each group, and if it is different than the one in the `gh-pages` branch, we expect the `version.json` file to be different?

Reply:
ok, that is even better :)
however, no small feat, so I'd rather do this in another PR, wdyt?

Reply:
sure

Reply:
removed the check, will come with #4182